solidity/src/FlowBridgedERC721.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract FlowBridgedERC721 is ERC721, ERC721URIStorage, ERC721Burnable, Ownable {
+    string public flowNFTAddress;
+    string public flowNFTIdentifier;
+
+    constructor(
+        address owner,
+        string memory name,
+        string memory symbol,
+        string memory _flowNFTAddress,
+        string memory _flowNFTIdentifier
+    ) ERC721(name, symbol) Ownable(owner) {
+        flowNFTAddress = _flowNFTAddress;
+        flowNFTIdentifier = _flowNFTIdentifier;
+    }
+
+    function safeMint(address to, uint256 tokenId, string memory uri) public onlyOwner {
+        _safeMint(to, tokenId);
+        _setTokenURI(tokenId, uri);
+    }
+
+    function tokenURI(uint256 tokenId) public view override(ERC721, ERC721URIStorage) returns (string memory) {
+        return super.tokenURI(tokenId);
+    }
+
+    function supportsInterface(bytes4 interfaceId) public view override(ERC721, ERC721URIStorage) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    function getFlowNFTAddress() public view returns (string memory) {
+        return flowNFTAddress;
+    }
+
+    function getFlowNFTIdentifier() public view returns (string memory) {
+        return flowNFTIdentifier;
+    }
+
+    function exists(uint256 tokenId) public view returns (bool) {
+        return _ownerOf(tokenId) != address(0);
+    }
+}
```

Comment:
OpenSea also uses `contractURI` to resolve contract-level metadata. Might be helpful to add (https://docs.opensea.io/docs/contract-level-metadata)

Reply:
Ah good callout